### ISSUE TITLE
Review/ensure that all Job fields are exposed

### ISSUE THREAD

**Issue by quartzmo** [GoogleCloudPlatform/google-cloud-ruby#1633]
See [Jobs](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#resource) for the full list of fields.

- [x] `userDefinedFunctionResources` (`query`)
- [x] `nullMarker` (`load`)
- [x] `autodetect` (`load`)
- [x] `labels` (`[Experimental]`)
- [x] `user_email`

Unsupported fields in the Jobs json schema:

```json
{
  "configuration": {
    "query": {
      "userDefinedFunctionResources": [
        {
          "resourceUri": string,
          "inlineCode": string
        }
      ],
      "preserveNulls": boolean,
      "schemaUpdateOptions": [
        string
      ]
    },
    "load": {
      "nullMarker": string,
      "autodetect": boolean,
      "schemaUpdateOptions": [
        string
      ]
    },
    "labels": {
      (key): string
    }
  },
  "user_email": string
}
```

**Comment by blowmage** [comment]
Most of those fields are marked `[Experimental]`. The direction we've been given is to _not_ implement those fields until they are no longer experimental, or we are given specific direction to add them. Are we sure all these fields are to be added?

**Comment by quartzmo** [comment]
@tswast 

The `schemaUpdateOptions` and `configuration.labels` properties are marked `[Experimental]` in the API docs. Should we expose them at this time?

**Comment by tswast** [comment]
Labels was explicitly requested to be supported in client libs by Jeremy on the BigQuery team.

I'm not certain about `schemaUpdateOptions`.

Related: is there a way to pass in options that aren't explicitly supported by the client libraries? I'm wondering about ways to better support alpha/beta customers.

**Comment by blowmage** [comment]
> Related: is there a way to pass in options that aren't explicitly supported by the client libraries? I'm wondering about ways to better support alpha/beta customers.

We've looked into before and there is no easy way. We have encouraged users who need this to skip the Google Cloud Ruby library and drop down to the Google API Client library.

**Comment by quartzmo** [comment]
>  is there a way to pass in options that aren't explicitly supported by the client libraries? 

We have issue #1638 for the discussion of supporting experimental features.

**Comment by quartzmo** [comment]
@tswast Since `Job.labels` is requested, should we also add support for a labels `filter` param to our method for listing jobs? Seems like scope creep for this deliverable, but without it Job labels won't have as much value.

**Comment by quartzmo** [comment]
@tswast Can you find out if `schemaUpdateOptions` should be exposed at this time?

**Comment by tswast** [comment]
Got confirmation from the backend team. The docs for `schemaUpdateOptions` are out-of-date. It's a GA feature.

It's considered a P2 item, so not a GA blocker for the client libs, but hopefully available soon after GA.

**Comment by quartzmo** [comment]
QueryJob `preserveNulls` is marked `[Deprecated]`, so I'm not going to expose it.

**Comment by quartzmo** [comment]
Closed by #1699, #1701, #1703.
